packages/repository/src/relations/relation.helpers.ts

```diff
@@ -0,0 +1,47 @@
+// Copyright IBM Corp. 2019. All Rights Reserved.
+// Node module: @loopback/repository
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import * as _ from 'lodash';
+import {Entity, EntityCrudRepository, Filter, Options, Where} from '..';
+
+/**
+ * Finds model instances that contain any of the provided foreign key values.
+ *
+ * @param targetRepository - The target repository where the model instances are found
+ * @param fkName - Name of the foreign key
+ * @param fkValues - One value or array of values of the foreign key to be included
+ * @param scope - Additional scope constraints (not currently supported)
+ * @param options - Options for the operations
+ */
+export async function findByForeignKeys<
+  Target extends Entity,
+  TargetRelations extends object,
+  ForeignKey extends StringKeyOf<Target>
+>(
+  targetRepository: EntityCrudRepository<Target, unknown, TargetRelations>,
+  fkName: ForeignKey,
+  fkValues: Target[ForeignKey][] | Target[ForeignKey],
+  scope?: Filter<Target>,
+  options?: Options,
+): Promise<(Target & TargetRelations)[]> {
+  // throw error if scope is defined and non-empty
+  // see https://github.com/strongloop/loopback-next/issues/3453
+  if (scope && !_.isEmpty(scope)) {
+    throw new Error('scope is not supported');
+  }
+
+  const value = Array.isArray(fkValues)
+    ? fkValues.length === 1
```

Comment:
Let's add a shortcut for empty array to return `[]` immediately. Include a test to cover this corner case too.

Reply:
If we pass in an empty array, it complains that `Type 'never[]' is not assignable to type 'number'.` (or whatever the type of the foreign key value is)

Reply:
You can do the following:
```ts
const fkIds: number[] = []; 
```